### PR TITLE
Add customOutputValidation

### PR DESCRIPTION
**Pull request by mjk90**
- Add customOutputValidation to allow custom validation rules in endpoints. 
- Update LWBAEndpoint with customOutputValidation to validate bid/mid/ask

### CODE REVIEW

File: src/adapter/types.ts [hunk 1]
```diff
@@ -127,6 +127,8 @@ export type CustomInputValidator<T extends EndpointGenerics> = (
   adapterSettings: T['Settings'],
 ) => AdapterError | undefined
 
+export type CustomOutputValidator = (output: Readonly<AdapterResponse>) => AdapterError | undefined
```
Comment (alecgard) [inline src/adapter/types.ts:130]: I think we could make this use the response type from the Transport, which should avoid needing to do the `as LwbaResponseDataFields['Data']` when it is defined

```suggestion
export type CustomOutputValidator<T extends ResponseGenerics> = (output: T) => AdapterError | undefined
```

**Review by alecgard — DISMISSED** [review]

File: src/index.ts [hunk 2]
```diff
@@ -234,6 +234,9 @@ async function buildRestApi(adapter: Adapter) {
           req as AdapterRequest<EmptyInputParameters>,
           reply as unknown as Promise<unknown>,
         )
+
+        adapter.validateOutput(req as AdapterRequest<EmptyInputParameters>, response)
```
Comment (karen-stepanyan) [inline src/index.ts:238]: Can we have tests for this ?

**Review by karen-stepanyan — DISMISSED** [review]
Missing tests, otherwise lgtm

File: src/adapter/lwba.ts [hunk 3]
```diff
@@ -59,6 +60,25 @@ export class LwbaEndpoint<T extends LwbaEndpointGenerics> extends AdapterEndpoin
...
+      if (!data.mid || !data.bid || !data.ask) {
+        throw new AdapterError({
+          statusCode: 500,
+          message: `Invariant voilation. LWBA response must contain mid, bid and ask prices.`,
+        })
+      }
+      if (data.mid < data.bid || data.mid > data.ask) {
+        throw new AdapterError({
+          statusCode: 500,
+          message: `Invariant voilation. Mid price must be between bid and ask prices.`,
```
Comment (martin-cll) [inline src/adapter/lwba.ts:76]: typos on the word `voilation`.

**Review by karen-stepanyan — APPROVED** [review]

**Pull request merged**